linera-core/src/lib.rs

```diff
@@ -4,6 +4,12 @@
 
 //! This module defines the core Linera protocol.
 
+#[cfg(all(target_arch = "wasm32", not(feature = "web")))]
+compile_error! {
+    "Compiling on `wasm32` is only supported on the browser.
+     To compile for the browser, pass the `web` feature."
```

Comment:
Optional nit: maybe "enable" would be a better word here?:

```suggestion
     To compile for the browser, enable the `web` feature."
```